samples/remote_attestation_eeid/README.md

```diff
@@ -0,0 +1,233 @@
+          uint8_t pem_key[512]; // public key information
+          uint8_t* remote_report;
+          size_t remote_report_size;
+      };
+      ```
+
+      Where:
+
+        - `pem_key` holds the public key that identifies enclave_a
+
+        - `remote_report` contains a remote report signed by the enclave platform for use in remote attestation
+
+   3. Ask enclave_b to attest (validate) enclave_a's remote report (remote_report from above)
+
+      This is done through the following call:
+      ```c
+      oe_call_enclave(enclave, "VerifyReportAndSetPKey", &args);
+      ```
+
+      In the enclave_b's implementation of `VerifyReportAndSetPKey`, it calls `oe_verify_report`, which will be described in the enclave section to handle all the platform specfic report validation operations (including PCK certificate chain checking). If successful the public key in `RemoteReportWithPKey.pem_key` will be stored inside the enclave for future use
+
+   4. Repeat step 2 and 3 for asking enclave_a to validate enclave_b
+
+   5. Free the resource used, including the host memory allocated by the enclaves and the enclaves themselves
+
+      For example:
+
+      ```c
+      oe_terminate_enclave(enclave_a);
+      oe_terminate_enclave(enclave_b);
+      ```
+
+### Authoring the Enclave
+
+#### Attesting an Enclave
+
+Attesting an enclave consists of three steps:
+
+##### 1) Generating an Enclave Report
+
+The enclave being attested first needs to generate a cryptographically strong proof of its identity that the challenger can verify. In the sample this is done by asking the SGX platform to generate a `remote report` signed by Intel via the `oe_get_report` method with `OE_REPORT_FLAGS_REMOTE_ATTESTATION` flag. The `remote report` can be verified by the `oe_verify_report` method on a different machine.
+
+An important feature of `oe_get_report` is that you can pass in application specific data as the `reportData` parameter to be signed into the report.
+
+- This is limited to 64 bytes in SGX. As illustrated in the sample, you sign arbitrarily large data into the report by first hashing it and then passing it to the `oe_get_report` method.
+
+- This is useful to bootstrap a secure communication channel between the enclave and the challenger.
+
+  - In this sample, the enclave signs the hash of an ephemeral public key into its report, which the challenger can then use to encrypt a response to it.
+
+  - Other usage examples for `reportData` might be to include a nonce, or to initiate Diffie-Helman key exchange.
+
+##### 2) Verifying the integrity of the Enclave Report
+
+Once the report is generated and passed to the challenger, the challenger can call `oe_verify_report` to validate the report originated from an Trust Execution Environment (TEE, in the case it's a valid SGX platform).
+
+In the context of Open Enclave on Intel SGX platform, a remote report is verified using the certificate chain issued by Intel which is only valid for SGX platforms.
```

Comment:
We're trying to align terminology with the IETF, so "report" is not a term we should use in docs.   We use "evidence" for information sent to a verifier, and "attestation results" for information from a verifier to a relying party.